### DISCUSSION TITLE
422 error but structure of data sent seems to be right

### DISCUSSION THREAD

**Discussion by Adib234** [fastapi/fastapi#6812] [Questions] [answered]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and didn't find an answer.
* [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [X] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

Go to /8000/docs and make a post request

### Environment

* OS: [macOS]:
* FastAPI Version [e.g. 0.3.0]:

To know the FastAPI version use:

```bash
0.63.0
```

* Python version:
3.9.4

### Additional context

Hello everyone I'm getting a 422 error from using the OpenAPI docs and here's my models and my function header

```python
class ImageProperty(BaseModel):
    text: str = Field(..., max_length=256)
    characteristics: str = Field(..., max_length=256)
    permissions: Optional[PermissionsEnum] = Field(PermissionsEnum.public)


class ImagesProperty(BaseModel):
    images: List[ImageProperty]


@router.post("/add/")
async def add_images(api_key: str, images_property: ImagesProperty, images_upload: List[UploadFile] = File(...)):
```

When I put the information into the interactive docs this is the curl request

```
curl -X 'POST' \
  'http://127.0.0.1:8000/add/?api_key=stringstring' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'images_property={
   "images":[
      {
         "text":"asdas",
         "characteristics":"asdfsd",
         "permissions":"public"
      }
   ]
}' \
  -F 'images_upload=@Screen Shot 2021-04-27 at 6.15.26 PM.png;type=image/png'
```

This is the response for the 422

```
{
  "detail": [
    {
      "loc": [
        "body",
        "images_property"
      ],
      "msg": "value is not a valid dict",
      "type": "type_error.dict"
    }
  ]
}
```

**Comment by Kludex** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/7353520/116790900-2287e300-aab7-11eb-8747-3b8c30724dba.png)

https://fastapi.tiangolo.com/tutorial/request-forms-and-files/

**Comment by Adib234** [comment]
So I saw the docs and it suggested to use forms and files in the same request and this is what I've changed.

```python
class ImageProperty(BaseModel):
    text: str = Field(..., max_length=256)
    characteristics: str = Field(..., max_length=256)
    permissions: Optional[PermissionsEnum] = Field(PermissionsEnum.public)


class ImagesProperty(BaseModel):
    api_key: str
    images: List[ImageProperty]


@router.post("/add/")
async def add_images(images_property: ImagesProperty = Form(...), images_upload: List[UploadFile] = File(...)):
```
Unfortunately I get the same error

**Comment by Adib234** [comment]
Are they also suggesting that I can't use a model in my request? 
> but you can't also declare Body fields that you expect to receive as JSON, as the request will have the body encoded using multipart/form-data instead of application/json.

**Comment by Adib234** [comment]
Okay yeah models are basically a request body so the quote above is what I suspected. I ended up just declaring all my properties in my model as lists of primitive values. This issue has been resolved. Thank you. 

Final result, no more models in headers, just query params
```python
async def add_images(api_key: str, t: List[str] = Query(..., alias="texts", title="Description for images"),
                     c: List[str] = Query(..., alias="characteristics",
                                          title="Tags for images"),
                     p: List[str] = Query(
                         'public', alias="permissions", title="Permissions of images"),
                     images_upload: List[UploadFile] = File(...)):
```

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! 👏  🙇 

Thanks for reporting back and closing the issue @Adib234  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
